hibernate-reactive-core/src/test/java/org/hibernate/PersistAllWithNoBatchSizeConfiguredTest.java

```diff
@@ -0,0 +1,149 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate;
+
+import java.util.Objects;
+import java.util.Set;
+
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.reactive.BaseReactiveTest;
+import org.hibernate.reactive.testing.SqlStatementTracker;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.vertx.junit5.Timeout;
+import io.vertx.junit5.VertxTestContext;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import jakarta.persistence.Version;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Timeout(value = 10, timeUnit = MINUTES)
+public class PersistAllWithNoBatchSizeConfiguredTest extends BaseReactiveTest {
+	private static SqlStatementTracker sqlTracker;
+
+	@Override
+	protected Set<Class<?>> annotatedEntities() {
+		return Set.of( GuineaPig.class );
+	}
+
+	@Override
+	protected Configuration constructConfiguration() {
+		Configuration configuration = super.constructConfiguration();
+
+		// Construct a tracker that collects query statements via the SqlStatementLogger framework.
+		// Pass in configuration properties to hand off any actual logging properties
+		sqlTracker = new SqlStatementTracker(
+				PersistAllWithNoBatchSizeConfiguredTest::filter,
+				configuration.getProperties()
+		);
+		return configuration;
+	}
+
+	@BeforeEach
+	public void clearTracker() {
+		sqlTracker.clear();
+	}
+
+	protected void addServices(StandardServiceRegistryBuilder builder) {
+		sqlTracker.registerService( builder );
+	}
+
+	private static boolean filter(String s) {
+		String[] accepted = { "insert ", "update ", "delete " };
+		for ( String valid : accepted ) {
+			if ( s.toLowerCase().startsWith( valid ) ) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	@Test
+	public void testBatchingWithStateless(VertxTestContext context) {
+		final GuineaPig[] pigs = {
+				new GuineaPig( 11, "One" ),
+				new GuineaPig( 22, "Two" ),
+				new GuineaPig( 33, "Three" ),
+				new GuineaPig( 44, "Four" ),
+				new GuineaPig( 55, "Five" ),
+				new GuineaPig( 66, "Six" ),
+		};
+		test( context, getMutinySessionFactory()
+				.withStatelessTransaction( s -> s.insertAll( 10, pigs ) )
```

Comment:
If it doesn't take too long, I think it would be helpful to test also test the other methods that accept the batch size as parameter: `deleteAll`, `updateAll`, and `refreshAll`

Reply:
Being a public API, I think we should also test with the `Stage` API.